contracts/deprecated/TendermintLightClient.sol

```diff
@@ -55,76 +55,7 @@ contract TendermintLightClient is ILightClient, System, IParamSubscriber {
-        require(height > initialHeight, "can't sync header before initialHeight");
-
-        uint64 preValidatorSetChangeHeight = latestHeight;
-        ConsensusState memory cs = lightClientConsensusStates[preValidatorSetChangeHeight];
-        for (; preValidatorSetChangeHeight >= height && preValidatorSetChangeHeight >= initialHeight;) {
-            preValidatorSetChangeHeight = cs.preValidatorSetChangeHeight;
-            cs = lightClientConsensusStates[preValidatorSetChangeHeight];
-        }
-        if (cs.nextValidatorSet.length == 0) {
-            preValidatorSetChangeHeight = cs.preValidatorSetChangeHeight;
-            cs.nextValidatorSet = lightClientConsensusStates[preValidatorSetChangeHeight].nextValidatorSet;
-            require(cs.nextValidatorSet.length != 0, "failed to load validator set data");
-        }
-
-        //32 + 32 + 8 + 32 + 32 + cs.nextValidatorSet.length;
-        uint256 length = 136 + cs.nextValidatorSet.length;
-        bytes memory input = new bytes(length + header.length);
-        uint256 ptr = Memory.dataPtr(input);
-        require(
-            encodeConsensusState(cs, preValidatorSetChangeHeight, ptr, length), "failed to serialize consensus state"
-        );
-
-        // write header to input
-        uint256 src;
-        ptr = ptr + length;
-        (src, length) = Memory.fromBytes(header);
-        Memory.copy(src, ptr, length);
-
-        length = input.length + 32;
-        // Maximum validator quantity is 99
-        bytes32[128] memory result;
-        /* solium-disable-next-line */
-        assembly {
-            // call validateTendermintHeader precompile contract
-            // Contract address: 0x64
-            if iszero(staticcall(not(0), 0x64, input, length, result, 4096)) { revert(0, 0) }
-        }
-
-        //Judge if the validator set is changed
-        /* solium-disable-next-line */
-        assembly {
-            length := mload(add(result, 0))
-        }
-        bool validatorChanged = false;
-        if ((length & (0x01 << 248)) != 0x00) {
-            validatorChanged = true;
-            ISystemReward(SYSTEM_REWARD_ADDR).claimRewards(msg.sender, rewardForValidatorSetChange);
-        }
-        length = length & 0xffffffffffffffff;
-
-        /* solium-disable-next-line */
-        assembly {
-            ptr := add(result, 32)
-        }
-
-        uint64 actualHeaderHeight;
-        (cs, actualHeaderHeight) = decodeConsensusState(ptr, length, !validatorChanged);
-        require(actualHeaderHeight == height, "header height doesn't equal to the specified height");
-
-        submitters[height] = msg.sender;
-        cs.preValidatorSetChangeHeight = preValidatorSetChangeHeight;
-        lightClientConsensusStates[height] = cs;
-        if (height > latestHeight) {
-            latestHeight = height;
-        }
-
-        emit syncConsensusState(height, preValidatorSetChangeHeight, cs.appHash, validatorChanged);
-
-        return true;
+        revert("deprecated");
```

Comment:
`decodeConsensusState` no longer needed,  simplify the `init` function please.

Reply:
fixed